### ISSUE TITLE
Navigation sidebar and Server sidebar not redering after leaving the server 

### ISSUE THREAD

**Issue by SaiSawant1** [AntonioErdeljac/next13-discord-clone#18]
After I leave a particular server, the page gets refreshed and I get redirected to the "/" page. However, the server sidebar and navigation sidebar are not rendered.
![Screenshot from 2023-09-10 10-37-29](https://github.com/AntonioErdeljac/next13-discord-clone/assets/92092643/4edbd20b-d9ef-4505-bfb3-d5c84e36b5ae)

**Comment by sethumadh** [comment]
Same thing happened to me too after deleting a server. I did a window.location.reload(). after router.refresh().   after the axios request. I am  sure there is a better right approach surely ,  anyway it is working at the moment.

**Comment by SaiSawant1** [comment]
I tried the same thing. I guess till someone fixes it we can use that workaround. Probably a nextJS bug

**Comment by caezd** [comment]
Think I found where it's coming from : 

Both these components use tailwind `className="hidden"` to handle mobile responsiveness and next/navigation redirect function seems to compile some of these classes inside `_next/static/css/app/(setup)/page.css`. On refresh, they go back inside `_next/static/css/app/layout.css`.

**Comment by sethumadh** [comment]
Did you manage to find a solution?

**Comment by caezd** [comment]
A real one, not really. 

So far there's #19 with some kind of solution, though I don't think adding `window.location.reload()` everywhere there's a redirect to be an optimal fix.

Mine neither, but for now I've gone the CSS road and added two classes that apply Tailwind's directives like so, pretty simple : 

```css
@layer components {
    .navigation-sidebar {
        @apply hidden md:flex h-full w-[72px] z-30 flex-col fixed inset-y-0;
    }
    .server-sidebar {
        @apply hidden md:flex h-full w-60 z-20 flex-col fixed inset-y-0;
    }
}
```

**Comment by Yvntrix** [comment]
any reason why this behavior occur? is it a nextjs bug?

**Comment by SaiSawant1** [comment]
> A real one, not really.
> 
> So far there's #19 with some kind of solution, though I don't think adding `window.location.reload()` everywhere there's a redirect to be an optimal fix (mine neither tbh).
> 
> For now I've gone the CSS road and added two classes that apply Tailwind's directives like so, pretty simple :
> 
> ```css
> @layer components {
>     .navigation-sidebar {
>         @apply hidden md:flex h-full w-[72px] z-30 flex-col fixed inset-y-0;
>     }
>     .server-sidebar {
>         @apply hidden md:flex h-full w-60 z-20 flex-col fixed inset-y-0;
>     }
> }
> ```

yeah, I think this approach is way better than window.location.reload();.
#19 was created by me. I will close that pull request since this is a way better approach however I want to know how you found out what was causing the bug

**Comment by AntonioErdeljac** [comment]
You can fix this by moving UploadThing's styles into global.css instead of react components

**Comment by dyljhd** [comment]
> You can fix this by moving UploadThing's styles into global.css at the end of the file instead of in the react component.
> 
> Step 1:
> 
> * Remove the import for upload-thing styles inside file-upload.tsx component
> * DELETE => import "@uploadthing/react/styles.css";
> 
> Step 2:
> 
> * Add the import at the bottom of the globals.css file instead
> 
> ```
> // globals.css
> ...
> @import "~@uploadthing/react/styles.css";
> ```
> 
> Step 3 (optional):
> 
> * Wrap the tailwind config with "withUt":
> 
> ```
> // tailwind.config.js
> 
> const { withUt } = require("uploadthing/tw");
> module.exports = withUt({
>    ...leave everything the same
> });
> ```

This has worked for me, so thank you for that. 
Would you mind explaining why moving these `UploadThing` styles fixes this issue?

**Comment by sethumadh** [comment]
Hi , Could you tell us how did you debug this issue?

**Comment by PSriVignesH** [comment]
Server sidebar not rerendering after editing or deleting the channels
